platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/EntityInformation.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security.samlp.impl;
+
+import static java.util.Objects.nonNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import javax.annotation.concurrent.Immutable;
+
+import org.apache.commons.lang.StringUtils;
+import org.opensaml.saml2.core.AuthnRequest;
+import org.opensaml.saml2.metadata.AssertionConsumerService;
+import org.opensaml.saml2.metadata.Endpoint;
+import org.opensaml.saml2.metadata.EntityDescriptor;
+import org.opensaml.saml2.metadata.KeyDescriptor;
+import org.opensaml.saml2.metadata.SPSSODescriptor;
+import org.opensaml.xml.schema.XSBase64Binary;
+import org.opensaml.xml.security.credential.UsageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+
+import ddf.security.samlp.SamlProtocol;
+import ddf.security.samlp.SamlProtocol.Binding;
+
+@Immutable
+public class EntityInformation {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EntityInformation.class);
+
+    private static final ImmutableSet<Binding> SUPPORTED_BINDINGS =
+            ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT);
```

Comment:
To be more clear. You've now indirectly coupled code that lives all the way up in security-core-impl with the implementation that lives in the IdP server.